### PR TITLE
WIP attempt to align the models more closely with the entity relations diagram

### PR DESCRIPTION
**Pull request by swortelboer**

### CODE REVIEW

File: loefsys/events/models/event_registation.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class EventRegistration(models.Model):
     user = models.ForeignKey(
         settings.AUTH_USER_MODEL,
         models.SET_NULL,
-        null=True
+        null=True,
```
Comment (SebastiaanRam) [inline loefsys/events/models/event_registation.py:21]: Comma here is not necessary, there are no additional vars

File: loefsys/events/models/event.py [hunk 2]
```diff
@@ -30,7 +35,21 @@ class Event(models.Model):
...
     # description = HTMLField(_("description"),)
 
-    # organiser (should use contenttypes for user, committee, yearclub)
+    organisers = models.ManyToManyField(
+        to=settings.AUTH_USER_MODEL,
+        related_name="event_organisers",
+        blank=True,
+    )
+    
+    contacts = models.ManyToManyField(
```
Comment (mark-boute) [inline loefsys/events/models/event.py:44]: This should be `organiser_contacts`, in the object create function we will check whether either `organiser_groups` or `organiser_contacts` contains at least one entry.

File: loefsys/events/models/event.py [hunk 3]
```diff
@@ -30,7 +35,21 @@ class Event(models.Model):
 
     # description = HTMLField(_("description"),)
 
-    # organiser (should use contenttypes for user, committee, yearclub)
+    organisers = models.ManyToManyField(
```
Comment (mark-boute) [inline loefsys/events/models/event.py:38]: this should be `organiser_groups`

File: loefsys/events/models/event.py [hunk 4]
```diff
@@ -30,7 +35,21 @@ class Event(models.Model):
...
+        to=settings.AUTH_USER_MODEL,
```
Comment (mark-boute) [inline loefsys/events/models/event.py:39]: should be to `groups`

File: loefsys/events/models/event.py [hunk 5]
```diff
@@ -30,7 +35,21 @@ class Event(models.Model):
...
+    
+    contacts = models.ManyToManyField(
+        to=settings.AUTH_USER_MODEL,
+        related_name="event_contact_persons",
+        blank=True,
+    )
+
+    group_organisers = GenericRelation('GroupOrganiser')
+    
+    group_contacts = GenericRelation('GroupContact')
```
Comment (mark-boute) [inline loefsys/events/models/event.py:52]: I do not think we will need these

File: loefsys/events/models/event.py [hunk 6]
```diff
@@ -195,9 +218,25 @@ def registration_closed(self):
         return (
             self.registration_end is not None and timezone.now() > self.registration_end
         )
+    
+    def clean(self):
+        super().clean()
+        # Custom validation to ensure at least one organiser or contact is set
+        if not self.organisers.exists() and not self.contacts.exists():
```
Comment (mark-boute) [inline loefsys/events/models/event.py:225]: should also be renamed, this should also be checked on create/edit as part of the object validation. So maybe this is not the place

File: loefsys/groups/models/membergroup.py [hunk 7]
```diff
@@ -12,7 +12,7 @@ def get_queryset(self):
 
 
 class MemberGroup(models.Model):
```
Comment (mark-boute) [inline loefsys/groups/models/membergroup.py:14]: Should now just be a group

File: loefsys/groups/models/membergroup.py [hunk 8]
```diff
@@ -29,7 +29,8 @@ class MemberGroup(models.Model):
     # )
 
     members = models.ManyToManyField(
-        settings.AUTH_USER_MODEL, through="groups.MemberGroupMembership"
+        settings.AUTH_USER_MODEL,
+        through="groups.MemberGroupMembership",
```
Comment (mark-boute) [inline loefsys/groups/models/membergroup.py:33]: `groups.GroupMemberShip`

File: loefsys/reservations/models/reservable/boat.py [hunk 9]
```diff
@@ -24,3 +27,26 @@ class Boat(Reservable):
         default="LOEFBIJTER",
     )
 
+    member_price = models.DecimalField(
```
Comment (mark-boute) [inline loefsys/reservations/models/reservable/boat.py:30]: The pricing logic must be in `Reservable`

File: loefsys/users/urls.py [hunk 10]
```diff
@@ -1,8 +1,8 @@
 from django.urls import path, include
 
-from members.views import ProfileDetailView, ProfileListView
+from users.views import ProfileDetailView, ProfileListView
 
-app_name = "members"
+app_name = "users"
```
Comment (mark-boute) [inline loefsys/users/urls.py:5]: we should discuss this naming, and maybe even move this logic to the base module of the system. As it not only covers users, but also external contacts.

**Review by mark-boute — CHANGES_REQUESTED** [review]

File: loefsys/events/models/event_registation.py (hunk 1 above)
Comment (mark-boute) [reply loefsys/events/models/event_registation.py:21]: I think the formatting tool does this

Comment (swortelboer) [reply loefsys/events/models/event_registation.py:21]: Believe I added these manually as I saw that sometimes we did and sometimes did not have trailing commas. I then left / added the optional trailing commas to be consistent and in line with the PEP 8 recommendation.

File: loefsys/users/urls.py (hunk 10 above)
Comment (swortelboer) [reply loefsys/users/urls.py]: Moved users to the base module of the system. Have yet to come up with a better name for it though.

File: users/apps.py [hunk 11]
```diff
@@ -4,11 +4,11 @@
...
 class UsersConfig(AppConfig):
     default_auto_field = "django.db.models.BigAutoField"
-    name = "loefsys.users"
+    name = "users"
     verbose_name = _("Users")
 
     def ready(self):
         try:
-            import loefsys.users.signals  # noqa F401
+            import users.signals  # noqa F401
```
Comment (mark-boute) [inline users/apps.py:12]: What is this?

**Review by mark-boute — COMMENTED** [review]
Nice work!!

File: users/apps.py (hunk 11 above)
Comment (mark-boute) [reply users/apps.py:12]: @SebastiaanRam any idea?

**Review by mark-boute — APPROVED** [review]

**Pull request merged**